courses/models.py

```diff
@@ -90,12 +91,21 @@ def catalog_image_url(self):
         )
 
 
-class Program(TimestampedModel, PageProperties):
+validate_url_path_field = RegexValidator(
+    r"[\s/]",
+    "This field is used to produce URL paths. It cannot contain spaces or forward slashes",
```

Comment:
I saved a new `Course` with a readable_id of `weird-test-course-1{foo}(!!@#$%)`, then created a Wagtail page for it.  When trying to view it, the URL did not work (404)  - http://xpro.odl.local:8053/courses/weird-test-course-1%7Bfoo%7D(!!@#$%)/

Doubtful anyone would deliberately create such a bizarre id, but maybe a stray special character might get in via user error.

Reply:
Good catch. This should use basically the same pattern as the URL resolver